deployment-scripts/jobs-deployment/tests/test_deploy_job.py

```diff
@@ -43,6 +43,7 @@ def test_should_start_sql_job_with_clean_state_if_not_running_and_no_previous_st
         self.flink_cli_runner.is_job_running = MagicMock(return_value=False)
         self.flink_cli_runner.get_job_id = MagicMock(return_value="test_job_id")
         self.flink_cli_runner.stop_with_savepoint = MagicMock()
+        self.flink_cli_runner._get_job_status = MagicMock(return_value="RUNNING")
```

Comment:
In vast majority of tests _get_job_status is irrelevant. You can move this line to `setup()` step and expose it (and overwrite if needed) in tests when the interaction is tested.